mobile/lib/app/features/integrations/view/integration_page.view.dart

```diff
@@ -0,0 +1,203 @@
+        integrationMediator.getIntegrations();
+
+    return Scaffold(
+      body: Padding(
+        padding: const EdgeInsets.all(8.0),
+        child: SafeArea(
+          child: Padding(
+            padding: const EdgeInsets.symmetric(horizontal: 8.0),
+            child: Column(
+              children: [
+                TriggoBanner(),
+                const SizedBox(height: 16.0),
+                Align(
+                  alignment: Alignment.centerLeft,
+                  child: Text(
+                    'Integrations',
+                    style: Theme.of(context).textTheme.titleLarge,
+                  ),
+                ),
+                const SizedBox(height: 4.0),
+                Expanded(
+                  child: Container(
+                    padding: const EdgeInsets.all(8.0),
+                    decoration: BoxDecoration(
+                      borderRadius: BorderRadius.circular(12.0),
+                      color: Colors.grey[200],
+                    ),
+                    child: Column(
+                      children: [
+                        Row(
+                          children: [
+                            Expanded(
+                              child: Padding(
+                                padding: const EdgeInsets.symmetric(
+                                    vertical: 8.0, horizontal: 4.0),
+                                child: IntegrationConnectionButton(),
+                              ),
+                            ),
+                          ],
+                        ),
+                        Expanded(
+                            child: IntegrationList(integrations: integrations)),
+                      ],
+                    ),
+                  ),
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class IntegrationList extends StatelessWidget {
+  const IntegrationList({super.key, required this.integrations});
+
+  final Future<List<Integration>> integrations;
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder<List<Integration>>(
+      future: integrations,
+      builder: (context, snapshot) {
+        return IntegrationListView(snapshot: snapshot);
+      },
+    );
+  }
+}
+
+class IntegrationListView extends StatelessWidget {
+  final AsyncSnapshot<List<Integration>> snapshot;
+
+  const IntegrationListView({required this.snapshot, super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    if (snapshot.connectionState == ConnectionState.waiting) {
+      return CircularProgressIndicator();
+    } else if (snapshot.hasError) {
+      return ErrorView(error: snapshot.error);
+    } else if (!snapshot.hasData || snapshot.data!.isEmpty) {
+      return const NoDataView();
+    } else {
+      return IntegrationListViewContent(integrations: snapshot.data!);
+    }
+  }
+}
+
+class ErrorView extends StatelessWidget {
+  final Object? error;
```

Comment:
I think you can remove this `?`
```suggestion
  final Object error;
```

Reply:
I can't because the type of `snapshot.error` is Object?

Reply:
So, if you SURE that snapshot.error is available when you trying to display the error, I think you can give error with `!`. Like this :
```
ErrorView(error: snapshot.error!)
```
I think this work

Reply:
You're right, didn't think about that !